kernel/src/process/unshare.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::sync::RwArc;
+
+use crate::{prelude::*, process::CloneFlags};
+
+/// Provides APIs for disassociating parts of the execution context
+pub trait ContextUnshareApi {
```

Comment:
Do you think it is a good idea to group `nsproxy`, `unshare`, and `uesr_ns` into a `process::namespace` module? This could help readers identify namespace-related modules quickly -- even if `process::namespace` does not provide any extra benefit of information hiding, abstraction, or encapsulation.